test/distributed/org/apache/cassandra/distributed/test/topology/DecommissionAvoidTimeouts.java

```diff
@@ -0,0 +1,240 @@
+import org.apache.cassandra.distributed.util.Coordinators;
+import org.apache.cassandra.distributed.util.QueryResultUtil;
+import org.apache.cassandra.distributed.util.byterewrite.StatusChangeListener;
+import org.apache.cassandra.distributed.util.byterewrite.Undead;
+import org.apache.cassandra.exceptions.ReadTimeoutException;
+import org.apache.cassandra.exceptions.WriteTimeoutException;
+import org.apache.cassandra.locator.DynamicEndpointSnitch;
+import org.apache.cassandra.locator.EndpointsForToken;
+import org.apache.cassandra.locator.IEndpointSnitch;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.locator.Replica;
+import org.apache.cassandra.locator.ReplicaCollection;
+import org.apache.cassandra.net.Verb;
+import org.apache.cassandra.service.StorageProxy;
+import org.apache.cassandra.utils.AssertionUtils;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public abstract class DecommissionAvoidTimeouts extends TestBaseImpl
+{
+    public static final int DECOM_NODE = 6;
+
+    @Test
+    public void test() throws IOException
+    {
+        try (Cluster cluster = Cluster.build(8)
+                                      .withRacks(2, 4)
+                                      .withInstanceInitializer(BB::install)
+                                      .withConfig(c -> c.with(Feature.GOSSIP)
+                                                        .set("transfer_hints_on_decommission", false)
+                                                        .set("severity_during_decommission", 100D))
+                                      .start())
+        {
+            fixDistributedSchemas(cluster);
+            cluster.schemaChange("CREATE KEYSPACE " + KEYSPACE + " WITH replication = {'class': 'NetworkTopologyStrategy', 'datacenter1': 3, 'datacenter2': 3}");
+            String table = KEYSPACE + ".tbl";
+            cluster.schemaChange("CREATE TABLE " + table + " (pk blob PRIMARY KEY)");
+
+            List<IInvokableInstance> dc1 = cluster.get(1, 2, 3, 4);
+            List<IInvokableInstance> dc2 = cluster.get(5, 6, 7, 8);
+            IInvokableInstance toDecom = dc2.get(1);
+            List<Murmur3Partitioner.LongToken> tokens = ClusterUtils.getLocalTokens(toDecom).stream().map(t -> new Murmur3Partitioner.LongToken(Long.parseLong(t))).collect(Collectors.toList());
+
+            for (Murmur3Partitioner.LongToken token : tokens)
+            {
+                ByteBuffer key = Murmur3Partitioner.LongToken.keyForToken(token);
+
+                toDecom.coordinator().execute("INSERT INTO " + table + "(pk) VALUES (?)", ConsistencyLevel.EACH_QUORUM, key);
+            }
+
+            CompletableFuture<Void> nodetool = CompletableFuture.runAsync(() -> toDecom.nodetoolResult("decommission").asserts().success());
+
+            StatusChangeListener.hooks(DECOM_NODE).leave.await();
+            cluster.filters().verbs(Verb.GOSSIP_DIGEST_SYN.id).drop();
```

Comment:
this makes sure that the LEFT gossip state never makes it to nodes.  I should also block the gossip shutdown message, that way the only thing coordinator knows is severity (and LEAVING)